src/m365/teams/commands/app/app-list.ts

```diff
@@ -30,59 +33,112 @@ class TeamsAppListCommand extends GraphItemsListCommand<TeamsApp> {
+      headers: {
+        accept: 'application/json;odata.metadata=none'
+      },
+      responseType: 'json'
+    };
 
-    this
-      .getAllItems(endpoint, logger, true)
-      .then((): void => {
-        if (args.options.output === 'json') {
-          logger.log(this.items);
+    return request
+      .get<{ value: Team[] }>(teamRequestOptions)
+      .then(response => {
+        const teamItem: Team | undefined = response.value[0];
+
+        if (!teamItem) {
+          return Promise.reject(`The specified team does not exist in the Microsoft Teams`);
         }
-        else {
-          if (args.options.teamId) {
-            logger.log((this.items as unknown as TeamsAppInstallation[]).map(i => {
-              return {
-                id: i.id,
-                displayName: i.teamsApp.displayName,
-                distributionMethod: i.teamsApp.distributionMethod
-              };
-            }));
-          }
-          else {
-            logger.log(this.items.map(i => {
-              return {
-                id: i.id,
-                displayName: i.displayName,
-                distributionMethod: i.distributionMethod
-              };
-            }));
-          }
+
+        if (response.value.length > 1) {
+          return Promise.reject(`Multiple Microsoft Teams teams with name ${args.options.teamName} found: ${response.value.map(x => x.id)}`);
         }
 
-        if (this.verbose) {
-          logger.log(chalk.green('DONE'));
+        return Promise.resolve(teamItem.id);
+      });
+  }
+
+  private getEndPointURL(args: CommandArgs): Promise<string> {
+    let endpoint: string = '';
+
+    return new Promise<string>((resolve: (endpoint: string) => void, reject: (error: string) => void): void => {
+      if (args.options.teamId || args.options.teamName) {
+        this
+          .getTeamId(args)
+          .then((teamId: string): void => {
+            endpoint = `${this.resource}/v1.0/teams/${encodeURIComponent(teamId)}/installedApps?$expand=teamsApp`;            
+
+            if (!args.options.all) {
+              endpoint += `&$filter=teamsApp/distributionMethod eq 'organization'`;
+            }
+
+            return resolve(endpoint);
+          })
+          .catch((err: any) => {
+            reject(err);
+          });
+      }
+      else {
+        endpoint = `${this.resource}/v1.0/appCatalogs/teamsApps`;
+
+        if (!args.options.all) {
+          endpoint += `?$filter=distributionMethod eq 'organization'`;
         }
 
-        cb();
-      }, (err: any): void => this.handleRejectedODataJsonPromise(err, logger, cb));
+        return resolve(endpoint);
+      }
+    });
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: (err?: any) => void): void {
+    this
+      .getEndPointURL(args)
+      .then((endpoint: string): void => {
+        this.getAllItems(endpoint, logger, true)
```

Comment:
We shouldn't nest promises, but instead attach them to the root chain to avoid unnecessary complexity.